pkg/operator/ceph/pool/radosnamespace/controller.go

```diff
@@ -243,6 +227,16 @@ func (r *ReconcileCephBlockPoolRadosNamespace) reconcile(request reconcile.Reque
 		r.updateStatus(r.client, namespacedName, cephv1.ConditionReady)
 		return reconcile.Result{}, nil
 	}
+
+	// Get CephCluster version
+	cephVersion, err := opcontroller.GetImageVersion(cephCluster)
+	if err != nil {
+		return opcontroller.ImmediateRetryResult, errors.Wrapf(err, "failed to fetch ceph version from cephcluster %q running in namespace %q", cephCluster.Name, cephCluster.Namespace)
```

Comment:
This will fail the reconcile if the version is not found. Can we limit the failure to just the case where the version is needed? For example, maybe we skip this version query if mirroring is not enabled on the rados namespace.